_posts/2022-09-19-blog-wpe-networking-overview.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+title: "WPE Networking Overview"
+tags: [blogpost]
+author: pgriffis
+permalink: /blog/04-wpe-networking-overview.html
+---
+
+# Overview of Networking in WPE
+
+At the heart of any brower engine is networking; Connecting with services and other users.  Unlike other engines WebKit approaches this more abstractly by leaving a large portion of the networking up to individual ports. This includes network protocols such as HTTP, WebSockets, and WebRTC. The upside to this approach is a higher level of integration with the system provided libraries and features so WebKit will behave similarly to other software on the platform often with more centralized configuration.
+
+## Networking Layers
+
+Due to this abstraction there are a few independent layers that make up the networking stack of WPE. In this post I'll break down what each layer accomplishes as well as brief insight into the codebases structure.
+
+### WebKit
+
+Before we get into the libraries used for WPE lets discuss WebKit itself. Despite abstracting out a lot of the protocol handling WebKit itself still needs to understand a lot of fundimentals of HTTP.
+
+WebCore (discussed in [WPE Overview](/blog/02-overview-of-wpe.html)) understands HTTP requests, headers, and cookies are as they are required to implement many higher level features. What it does not do is the network operations, most parsing, or on-disk storage for them. In the codebase these are represented by `ResourceRequest` and `ResourceResponse` objects mapping to general HTTP functionality.
+
+#### NetworkProcess
+
+A core part of modern web engine security is the multi-process model. Each website runs in its own isolated process that does not have network access to defend from exploits. In order to allow for network access they must talk over IPC to a dedicated NetworkProcess, one per browser instance typically. The NetworkProcess recieves a `ResourceRequest`, creates a `NetworkDataTask` out out if to download the data, and responds with a `ResourceResponse` to the WebProcess.
```

Comment:
* "recieves" -> "receives"
  * "creates a `NetworkDataTask` out out if to download the data" -> do you mean ""creates a `NetworkDataTask` **out** if **it needs** to download the data" ?